lib/atlas/sparse_graph_query.rb

```diff
@@ -0,0 +1,36 @@
+module Atlas
+  class SparseGraphQuery
+    include ActiveDocument
+
+    DIRECTORY = 'sparse_graph_queries'
+    VALID_NAME = /^[\w_@-]+\+[\w_]+$/
+
+    attribute :graph_part, String
+    attribute :graph_method, String
```

Comment:
Correct me if I'm wrong, but I don't think `graph_part` or `graph_method` need to be declared as `attribute`s. This would cause those values to be saved inside the document, which is redundant as they're part of the filename. 

`Edge` defines it's filename-based attributes like so:

```ruby
attr_reader :supplier
attr_reader :consumer
attr_reader :carrier

def supplier=(supplier)
  @supplier = supplier && supplier.to_sym
end

# etc ...
```